src/Microsoft.Extensions.Caching.Memory/MemoryCache.cs

```diff
@@ -122,7 +136,9 @@ private void SetEntry(CacheEntry entry)
                 priorEntry.SetExpired(EvictionReason.Replaced);
```

Comment:
I keep the size operations beside the actual calls to remove from the underlying concurrent dictionary. In this case, that's the ScanForExpiredItems call. I think we decided not to remove the entry right away to ensure the value can never be null on replacement hence the awkward logic here. I don't think it's safe to deduct the size here.